test/lightning_client_mock.go

```diff
@@ -132,3 +133,23 @@ func (h *mockLightningClient) SignMessage(ctx context.Context, msg []byte) (
 
 	return h.lnd.Signature, nil
 }
+
+func (h *mockLightningClient) VerifyMessage(ctx context.Context, msg []byte,
+	sig string) (bool, *btcec.PublicKey, error) {
+
+	pubkeyBytes, err := hex.DecodeString(h.lnd.NodePubkey)
```

Comment:
If you start decoding here, maybe it is better to make the field `route.Vertex` and using `String()` when needed.

Reply:
The pubkey parsing is not necessary now since we don't return it in `VerifyMessage` anymore.